### PR TITLE
after creating a new prospect, the user will be directed to the page …

### PR DESCRIPTION
**Pull request by GeorgeMocanu23**
…where he can add notes

### CODE REVIEW

File: components/prospects/CreateProspect.tsx [hunk 1]
```diff
@@ -29,6 +30,15 @@ function CreateProspect() {
           severity: 'success',
           text: 'Prospect successfully created!'
         })
+        fetch('/api/prospect/search?input=' + prospect.mail, {
```
Comment (S7012MY) [inline components/prospects/CreateProspect.tsx]: can we do it without calling fetch a second time?

**Review by S7012MY — CHANGES_REQUESTED** [review]

File: components/prospects/CreateProspect.tsx (hunk 1 above)
Comment (GeorgeMocanu23) [reply components/prospects/CreateProspect.tsx]: Surely the user can be redirected by another method, but today, all day I did not find a solution that could help me.
I notice that the 'SearchProspect' component renders 'CreateProspect' but without any parameters and I think that 'CreateProspect' should receive as a parameter the last created 'id', and then I can use it to complete the correct route.

Now I'm not sure if 'CreateProspect' should be rendered here or from another component.

A little hint would help me a lot.

File: cypress/integration/prospects/list.js [hunk 2]
```diff
@@ -72,14 +72,17 @@ describe('Create prospect', () => {
...
-    // reload the page and check if the newly created prospect appears in the
-    // list, also we need to scroll all the way down 3 times to make the 21st
+    // check if the user is directed to the newly created prospect
     // prospect visible
-    cy.reload()
+    cy.url().should('eq', 'http://localhost:3000/prospects/profile/21')
+    cy.contains('test@wellcode.com')
+    cy.contains("Notes")
+
+    // list, also we need to scroll all the way down 3 times to make the 21st
```
Comment (keepcoding-tech) [inline cypress/integration/prospects/list.js]: Let's rewrite the comment so it actually makes sense

File: cypress/integration/prospects/list.js [hunk 3]
```diff
@@ -72,14 +72,17 @@ describe('Create prospect', () => {
     cy.get('button[id="create"').click()
     cy.contains('Prospect successfully created!')
 
-    // reload the page and check if the newly created prospect appears in the
-    // list, also we need to scroll all the way down 3 times to make the 21st
+    // check if the user is directed to the newly created prospect
     // prospect visible
```
Comment (keepcoding-tech) [inline cypress/integration/prospects/list.js]: You also need to delete this comment

**Review by keepcoding-tech — CHANGES_REQUESTED** [review]

File: components/prospects/CreateProspect.tsx (hunk 1 above)
Comment (keepcoding-tech) [reply components/prospects/CreateProspect.tsx]: I think the quickest way would be to modify the API end-point to return the `id` of the prospect too.

File: components/prospects/CreateProspect.tsx [hunk 4]
```diff
@@ -29,6 +30,7 @@ function CreateProspect() {
           severity: 'success',
```
Comment (keepcoding-tech) [inline components/prospects/CreateProspect.tsx:30]: I don't think we still need to set the message to `success` if we redirect the user anyway.

File: pages/api/prospect/create.tsx [hunk 5]
```diff
@@ -34,15 +34,15 @@ export default async (req: NextApiRequest, res: NextApiResponse) => {
   }
 
   // insert the new prospect into the database 
-  await prisma.prospects.create({ data: {
+  const createdProspect = await prisma.prospects.create({ data: {
```
Comment (keepcoding-tech) [inline pages/api/prospect/create.tsx]: Just `prospect` I think is a good enough name

**Review by keepcoding-tech — CHANGES_REQUESTED** [review]

File: cypress/integration/prospects/list.js [hunk 6]
```diff
@@ -72,14 +72,18 @@ describe('Create prospect', () => {
...
+    cy.contains("Notes")
+
+    // should go to the prospects list and scroll all the way down 3 times
+    // to check if the created and opened prospect appears in the list
+    cy.contains('Prospects').click()
+    cy.url().should('eq', 'http://localhost:3000/prospects/list')
     cy.scrollTo('bottom').contains('Prospect 10')
     cy.scrollTo('bottom').contains('Prospect 20')
     cy.scrollTo('bottom').contains('test name')
-    cy.contains('test@wellcode.com')
```
Comment (S7012MY) [inline cypress/integration/prospects/list.js:82]: we should still check that the newly created prospect still appears in the list

**Review by S7012MY — CHANGES_REQUESTED** [review]

**Review by keepcoding-tech — APPROVED** [review]

**Review by S7012MY — APPROVED** [review]

**Pull request merged**